hbase-server/src/main/java/org/apache/hadoop/hbase/replication/HBaseReplicationEndpoint.java

```diff
@@ -193,12 +194,16 @@ private void reloadZkWatcher() throws IOException {
       if (zkw != null) {
         zkw.close();
       }
-      zkw =
-        new ZKWatcher(ctx.getConfiguration(), "connection to cluster: " + ctx.getPeerId(), this);
+      zkw = new ZKWatcher(ctx.getConfiguration(), "connection to cluster: " + ctx.getPeerId(), this,
+        getZKClientConfig());
       zkw.registerListener(new PeerRegionServerListener(this));
     }
   }
 
+  protected ZKClientConfig getZKClientConfig() {
```

Comment:
Better add some comments here to describe why we want to add this method? It is not used directly in the hbase code base, so maybe later other developers may decided to remove it...

Reply:
Thanks for pointing it out, let me add an explanation.